Tests/TestingApps/PaywallsTester/PaywallsTester/Views/AppContentView.swift

```diff
@@ -139,16 +134,75 @@ struct AppContentView: View {
         }
     }
 
-    private var isPurchasesConfigured: Bool {
-        return self.customerInfoStream != nil
+    private func reconfigure(for mode: Configuration.Mode) {
+        configuration.reconfigure(for: mode)
+        self.observeCustomerInfoStream()
     }
 
+    private func observeCustomerInfoStream() {
+        self.customerInfoTask?.cancel()
+        self.customerInfoTask = Task {
+            if Purchases.isConfigured {
+                for await info in Purchases.shared.customerInfoStream {
+                    self.customerInfo = info
+                    self.showingDefaultPaywall = self.showingDefaultPaywall && info.activeSubscriptions.count == 0
+                }
+            }
+        }
+    }
+
+    private func descriptionForCurrentMode() -> String {
+
+        switch self.configuration.currentMode {
+        case .custom:
+            return "the API set locally in Configuration.swift"
+        case .testing:
+            return "the Paywalls Tester app in RevenueCat Dashboard"
+        case .demos:
+            return "Demos"
+        }
```

Comment:
Nit: I'd make this an extension on `Configuration.Mode` and just use `self.configuration.currentMode.description` instead of creating a separate function

Reply:
I had it like that initially, but it felt pretty awkward because it's not really a generic description of the object that I have here, it's the copy for a label that includes more stuff